app/src/main/java/org/simple/clinic/di/network/RetrofitModule.kt

```diff
@@ -38,12 +40,18 @@ class RetrofitModule {
   @Provides
   @AppScope
   @Named("for_country")
-  fun retrofit(commonRetrofitBuilder: Retrofit.Builder): Retrofit {
-    val baseUrl = BuildConfig.API_ENDPOINT
-    val currentApiVersion = "v3"
+  fun retrofit(
+      country: Country,
+      commonRetrofitBuilder: Retrofit.Builder
+  ): Retrofit {
+    // Since the endpoint is not under our control, and is defined at the server level,
+    // this is a safety check that will generate the right endpoint regardless of whether the
+    // endpoint defined in the manifest has a trailing slash or not.
+    val baseUrl = country.endpoint.toString().removeSuffix("/")
```

Comment:
Pruning the endpoint looks like an overkill.
The endpoint should either have a slash or not and this should be a contract. 
I don't think we need a fallback for this.

Reply:
@ragunathjawahar and I discussed this at length. Since we do not have any way to enforce this contract from our side and it's dependent on humans to configure, we decided to add this check.

In addition, the ones who are configuring this endpoint are people who may not be aware that Retrofit requires the endpoint to end in a trailing slash, so this is currently some tribal knowledge that it might be prudent to account for.

Reply:
Question: When we are talking about people configuring the endpoint, are they people outside the Simple team?

Reply:
Not at the moment. But it's still tribal knowledge even within the mobile team since this is something that is not done frequently.

Reply:
The endpoint being configured incorrectly points to a different problem. This can happen irrespective of the slash. The app should still be able to recover from this issue. If not, we should talk to server devs about documenting this maybe.
I am merging this PR for now. Meanwhile let's discuss this elsewhere.